src/main/java/com/google/devtools/build/lib/bazel/repository/cache/RepositoryCache.java

```diff
@@ -237,19 +220,11 @@ public synchronized void put(
     FileSystemUtils.moveFile(tmpName, cacheValue);
 
     if (!Strings.isNullOrEmpty(canonicalId)) {
-      byte[] canonicalIdBytes = canonicalId.getBytes(UTF_8);
-      String idHash = keyType.newHasher().putBytes(canonicalIdBytes).hash().toString();
-      OutputStream idStream = cacheEntry.getRelative(ID_PREFIX + idHash).getOutputStream();
-      idStream.write(canonicalIdBytes);
-      idStream.close();
+      String idHash = keyType.newHasher().putBytes(canonicalId.getBytes(UTF_8)).hash().toString();
+      FileSystemUtils.touchFile(cacheEntry.getRelative(ID_PREFIX + idHash));
```

Comment:
hmmm... your PR description didn't mention this change (from writeFile to just touchFile). Could you briefly describe why this is done / why the old thing is unnecessary?

Reply:
I added an explanation to the description.